### DISCUSSION TITLE
direnv and .envrc can specify version?

### DISCUSSION THREAD

**Discussion by roccho-dev** [jdx/mise#1258] [Q&A]
https://github.com/jdx/rtx?tab=readme-ov-file#direnv

After reading here, I recognized and configured the current state of rtx as follows
* I installed `rtx install node@lts` first, which resulted in ver. 20 being installed
* rtx activate conflicts with direnv
* In fact, in my environment, `rtx activate` does not work when .envrc is automatically loaded
* So, I followed the above readme, set up `use_rtx.sh` and put the runtime version `RTX_NODE_VERSION=19.0.0` in the `.envrc`.

However, in the directory where `.envrc` is loaded, the environment variable is updated to ver. 19, but `node -v` outputs ver. 20.

Am I forgetting something?

**Comment by Marukome0743** [comment]
For current versions of mise, the corresponding environment variable is `MISE_NODE_VERSION` rather than the former `RTX_NODE_VERSION`.

The legacy direnv integration can be configured with:

```sh
mise direnv activate > ~/.config/direnv/lib/use_mise.sh
```

and the following `.envrc`:

```sh
export MISE_NODE_VERSION=19.0.0
use mise
```

I tested this on the current version of mise with direnv 2.37.1 and an actual Node.js 19.0.0 installation. Inside `direnv exec`, `node` was resolved from mise’s `node/19.0.0/bin` directory and reported:

```text
v19.0.0
```

However, it is important to note that the `use mise` integration is now deprecated and no longer officially supported. The recommended setup is to activate mise normally in the shell and put the tool version in `mise.toml`, for example:

```toml
[tools]
node = "19.0.0"
```

direnv can then be reserved for environment variables that mise is not managing. The current compatibility approach is documented here:

https://mise.en.dev/direnv.html

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
